mne/io/eeglab/eeglab.py

```diff
@@ -488,16 +488,24 @@ def __init__(self, input_fname, events=None, event_id=None, tmin=0,
                      ' of the first event will be retained.')
 
             # now fill up the event array
-            events = np.zeros((eeg.trials, 3), dtype=int)
-            for idx in range(0, eeg.trials):
-                if idx == 0:
-                    prev_stim = 0
-                elif (idx > 0 and
-                        event_latencies[idx] - event_latencies[idx - 1] == 1):
-                    prev_stim = event_id[event_name[idx - 1]]
-                events[idx, 0] = event_latencies[idx]
-                events[idx, 1] = prev_stim
-                events[idx, 2] = event_id[event_name[idx]]
+            if len(epochs) > 1:
+                events = np.zeros((len(epochs), 3), dtype=int)
+                for idx in range(0, len(epochs)):
+                    if idx == 0:
+                        prev_stim = 0
+                    elif (idx > 0 and
+                          event_latencies[idx] -
+                          event_latencies[idx - 1] == 1):
+                        prev_stim = event_id[event_name[idx - 1]]
+                    events[idx, 0] = event_latencies[idx]
+                    events[idx, 1] = prev_stim
+                    events[idx, 2] = event_id[event_name[idx]]
+            else:
+                # create placeholder event_id and events here to support
+                # single-epoch files
+                events = np.zeros((1, 3), dtype=int)
+                events[0, 2] = 1
+                event_id = {'1': 1}
```

Comment:
I am confused by the fix you suggest. TO me you should have eeg.trials == 1 and going through the for loop should work

Reply:
Because if there's only one trial the event-related variables won't exist, so using `event_id` wouldn't work. In EEGLAB, I don't think there's a distinct difference between "epochs" or "raw" data like MNE has, so this is one of the reasons I suggested throwing an exception earlier.